### PR TITLE
fea(hlapi): allow use of multibit for integers

### PR DESCRIPTION
**Pull request by tmontaigu**
### Check-list:

* [x] Tests for the changes have been added (for bug fixes / features)
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Relevant issues are marked as resolved/closed, related issues are linked in the description
* [x] Check for breaking changes (including serialization changes) and add them to commit message following the conventional commit [specification][conventional-breaking]

[conventional-breaking]: https://www.conventionalcommits.org/en/v1.0.0/#commit-message-with-description-and-breaking-change-footer

### CODE REVIEW

File: tfhe/src/c_api/high_level_api/config.rs [hunk 1]
```diff
@@ -80,9 +80,10 @@ pub unsafe extern "C" fn config_builder_enable_custom_integers(
     catch_panic(|| {
         check_ptr_is_non_null_and_aligned(builder).unwrap();
 
+        let params: crate::shortint::ClassicPBSParameters = shortint_block_parameters.into();
```
Comment (IceTDrinker) [inline tfhe/src/c_api/high_level_api/config.rs:83]: not generic enough to allow multi bit with the C API ?

File: tfhe/src/high_level_api/integers/keys.rs [hunk 2]
```diff
@@ -35,24 +35,28 @@ impl IntegerConfig {
...
     pub fn enable_wopbs(&mut self) {
         let block_parameter = self
             .block_parameters
-            .get_or_insert(crate::shortint::parameters::PARAM_MESSAGE_2_CARRY_2_KS_PBS);
+            .get_or_insert(crate::shortint::parameters::PARAM_MESSAGE_2_CARRY_2_KS_PBS.into());
 
-        let wopbs_block_parameters = match block_parameter.encryption_key_choice {
+        let wopbs_block_parameters = match block_parameter.encryption_key_choice() {
             EncryptionKeyChoice::Big => crate::shortint::parameters::parameters_wopbs_message_carry::WOPBS_PARAM_MESSAGE_2_CARRY_2_KS_PBS,
             EncryptionKeyChoice::Small => crate::shortint::parameters::parameters_wopbs_message_carry::WOPBS_PARAM_MESSAGE_2_CARRY_2_KS_PBS,
```
Comment (IceTDrinker) [inline tfhe/src/high_level_api/integers/keys.rs:61]: just looks a bit odd, but I don't think we have wopbs params for small params, so maybe error out on the small choice for now ?

File: tfhe/src/c_api/high_level_api/config.rs (hunk 1 above)
Comment (tmontaigu) [reply tfhe/src/c_api/high_level_api/config.rs:83]: Its because of https://github.com/zama-ai/tfhe-rs-internal/issues/140

**Review by IceTDrinker — APPROVED** [review]
Missing a fix on commit message otherwise looks good !

**Pull request merged**